### ISSUE TITLE
bug(api): Need to correct RRule formats.

### ISSUE THREAD

**Issue by elliotcourant** [monetr/monetr#1487]
Right now when we store the RRule we are truncating parts of it. This was fine originally but now is causing an issue where we don't have the date started for the rule. This date is incredibly important for forecasting correctly.

When we convert the rule to a string on the frontend:

https://github.com/monetr/monetr/blob/e792ad02ebae029d3ee7bad8e7bfe6f4a56b7421/ui/components/Recurrence/Recurrence.ts#L13-L15

We aren't including the date started and we remove the RRULE string that would separate that data.

Then on the backend we actually have a hard coded DTStart?

https://github.com/monetr/monetr/blob/e792ad02ebae029d3ee7bad8e7bfe6f4a56b7421/pkg/models/rule.go#L32-L36

Which works for some recurrence rules, mostly the happy path ones I've been testing with. But definitely will not work once more people start to use this.

---

Basically we need to make sure that the dtstart gets incorporated into the rule that we save in the database. Right now this is stored on the object itself. 

https://github.com/monetr/monetr/blob/e792ad02ebae029d3ee7bad8e7bfe6f4a56b7421/pkg/models/funding_schedule.go#L31

But that isn't sustainable.

**Comment by elliotcourant** [comment]
This will involve a DB migration, we would need to read all of the existing funding and spending object. Determine their date started. Then rewrite them all back to the DB with a corrected RRule.

This migration probably will not be backwards compatible. We could maybe maintain two rule columns temporarily if it absolutely needs to be.

**Comment by elliotcourant** [comment]
https://github.com/monetr/monetr/pull/1532 is kicking off the work on this

**Comment by elliotcourant** [comment]
Resolved by https://github.com/monetr/monetr/pull/1532
